snapcraft/plugins/catkin.py

```diff
@@ -919,3 +961,58 @@ def _get_highest_version_path(path):
         raise RuntimeError('nothing found in {!r}'.format(path))
 
     return paths[-1]
+
+
+class _Wstool:
+    def __init__(self, ros_package_path, wstool_path, ubuntu_sources, project):
+        self._ros_package_path = ros_package_path
+        self._ubuntu_sources = ubuntu_sources
+        self._wstool_path = wstool_path
+        self._wstool_install_path = os.path.join(wstool_path, 'install')
+        self._project = project
+
+    def setup(self):
+        os.makedirs(self._wstool_install_path, exist_ok=True)
+
+        # wstool isn't a dependency of the project, so we'll unpack it
+        # somewhere else, and use it from there.
```

Comment:
Not really, no. This is similar to the compilers and catkin bits: they (in most cases) come from the ROS repos, not Ubuntu's. We can't assume those repos are setup on the host machine, and we certainly don't want to add them. So we don't install them at all: we fetch them off to the side and use them. They don't end up in the snap, and they don't provide any libs, so no need to track them at all, I'd say.